src/helpers.rs

```diff
@@ -1434,3 +1447,35 @@ pub(crate) fn windows_check_buffer_size((success, len): (bool, u64)) -> u32 {
         u32::try_from(len).unwrap()
     }
 }
+
+pub trait WriteIoResult {
```

Comment:
Please add a doc comment. Also the trait name is kind of confusing, maybe a doc comment will help coming up with a better one.